### PR TITLE
[js/WebGPU] Support int32 Transpose in WebGPU

### PR DESCRIPTION
**Pull request by hariharans29**
### Description
Support (u)int32 transpose in JSEP/WebGPU as it is required for Segment Anything Encoder per operator tracking table: https://github.com/microsoft/onnxruntime/issues/15952.

### Motivation and Context
Improve WebGPU op coverage

### CODE REVIEW

**Comment by axinging** [comment]
Maybe you forget to add test case in js/web/test/suite-test-list.jsonc? 
We have implemented  transpose_int32.jsonc here:
https://github.com/xhcao/onnxruntime/blob/webgpu_transpose_int32/js/web/test/data/ops/transpose_int32.jsonc

File: js/web/lib/wasm/jsep/webgpu/ops/transpose.ts [hunk 1]
```diff
@@ -23,8 +23,9 @@ const validateInputs = (inputs: readonly TensorView[]): void => {
     throw new Error('Transpose requires 1 input.');
   }
 
-  if (inputs[0].dataType !== DataType.float) {
-    throw new Error('input should be float tensor');
+  if (inputs[0].dataType !== DataType.float && inputs[0].dataType !== DataType.int32 &&
```
Comment (guschmue) [inline js/web/lib/wasm/jsep/webgpu/ops/transpose.ts:26]: nit: don't think this checkk is needed since the registration already takes care of it.

**Review by guschmue — DISMISSED** [review]

**Comment by guschmue** [comment]
ci is failing because of test*json formatting - I think there is a fix from Yulong being merged.

File: js/web/lib/wasm/jsep/webgpu/ops/transpose.ts (hunk 1 above)
Comment (hariharans29) [reply js/web/lib/wasm/jsep/webgpu/ops/transpose.ts:26]: I thought so too - but kept it consistent with how it was previously.

**Review by guschmue — APPROVED** [review]

**Pull request merged**